symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/config/BdkServiceConfig.java

```diff
@@ -35,13 +35,15 @@
 import com.symphony.bdk.template.freemarker.FreeMarkerEngine;
 
 import org.apiguardian.api.API;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
 
 /**
  * Injection of Core services within the Spring application context.
  */
 @API(status = API.Status.EXPERIMENTAL)
+@ConditionalOnBean(name = "botSession")
```

Comment:
Why we don't inject services? As a developer I'd expect : 

```java
@Autowired 
private MessageService messageService; 

@Autowired 
private OboAuthenticator oboAuthenticator;

public void myMethod() {

    this.messageService.send(...) // throws IllegalStateException as botSession not configured

    final AuthSession oboSession = this.oboAuthenticator.authenticateByUsername("user.name");
    this.messageService.obo(oboSession).send(...) // that works 
}
```

Reply:
`botSession` has an annotation `@ConditionalOnProperty("bdk.bot.username")`. So if no bot service account is configured, Spring Boot initialization will fail.

Reply:
I guess we can do this: 
```
@Bean
  @ConditionalOnMissingBean
  public AuthSession botSession(AuthenticatorFactory authenticatorFactory) {
    try {
      return authenticatorFactory.getBotAuthenticator().authenticateBot();
    } catch (AuthUnauthorizedException | AuthInitializationException e) {
      return null;
    }
  }
```
then we can instantiate services beans with null session.

Reply:
New sets of service beans are created with a null authSession. So @symphony-thibault example above will work and will throw a NullPointerException in case we try to use a non-OBO usecase.

Reply:
@symphony-elias is it possible/easy to throw something else that NPE? I'm more thinking about an `new IllegalStateException("Bot service account is not configured")`, that would be clearer for developers